test/parallel/xyk-pallet.API.poolVestedToken.test.ts

```diff
@@ -0,0 +1,360 @@
+    await createPoolAndVestingToken(true, toBN("1", 20), new BN(100));
+
+    const burnLiudityResult = await burnLiquidity(
+      testUser1.keyRingPair,
+      MGA_ASSET_ID,
+      createdToken,
+      new BN(150000)
+    );
+    expect(getEventResultFromMangataTx(burnLiudityResult).state).toEqual(
+      ExtrinsicResult.ExtrinsicFailed
+    );
+    expect(getEventResultFromMangataTx(burnLiudityResult).data).toContain(
+      "NotEnoughAssets"
+    );
+
+    const transferAssetResult = await transferAsset(
+      testUser1.keyRingPair,
+      liquidityID,
+      sudo.keyRingPair.address.toString(),
+      new BN(150000)
+    );
+    expect(getEventResultFromMangataTx(transferAssetResult).state).toEqual(
+      ExtrinsicResult.ExtrinsicFailed
+    );
+    expect(getEventResultFromMangataTx(transferAssetResult).data).toContain(
+      "LiquidityRestrictions"
+    );
+  });
+
+  test("xyk-pallet- check that unlocking tokens are free to use when the relock time is done", async () => {
+    const api = getApi();
+
+    await createPoolAndVestingToken(true, toBN("1", 20), new BN(100));
+
+    const UserBalanceBeforeAmount = await api.query.tokens.accounts(
+      testUser1.keyRingPair.address,
+      liquidityID
+    );
+
+    expect(
+      UserBalanceBeforeAmount.free.sub(UserBalanceBeforeAmount.frozen)
+    ).bnEqual(new BN(0));
+
+    const unlockSomeVestedToken = await unlockVestedToken(
+      testUser1,
+      liquidityID
+    );
+    expect(getEventResultFromMangataTx(unlockSomeVestedToken).state).toEqual(
+      ExtrinsicResult.ExtrinsicSuccess
+    );
+
+    const UserBalanceAfterUnlockingAmount = await api.query.tokens.accounts(
+      testUser1.keyRingPair.address,
+      liquidityID
+    );
+    expect(
+      UserBalanceAfterUnlockingAmount.free.sub(
+        UserBalanceAfterUnlockingAmount.frozen
+      )
+    ).bnGt(new BN(0));
+
+    //@ts-ignore
+    const maxInstantBurnAmount = await api.rpc.xyk.get_max_instant_burn_amount(
+      testUser1.keyRingPair.address,
+      liquidityID
+    );
+
+    const burnUnlockedToken = await burnLiquidity(
+      testUser1.keyRingPair,
+      MGA_ASSET_ID,
+      createdToken,
+      maxInstantBurnAmount
+    );
+    expect(getEventResultFromMangataTx(burnUnlockedToken).state).toEqual(
+      ExtrinsicResult.ExtrinsicSuccess
+    );
+
+    const UserBalanceAfterBurningAmount = await api.query.tokens.accounts(
+      testUser1.keyRingPair.address,
+      liquidityID
+    );
+    expect(
+      UserBalanceAfterBurningAmount.free.sub(
+        UserBalanceAfterBurningAmount.frozen
+      )
+    ).bnEqual(new BN(0));
+
+    expect(
+      UserBalanceBeforeAmount.frozen.sub(UserBalanceAfterBurningAmount.frozen)
+    ).bnGt(new BN(0));
+  });
+
+  test("xyk-pallet- check that all unlocking vesting tokens can be burned", async () => {
+    const api = getApi();
+    const poolLockedValue = toBN("1", 20);
+    const poolBlockValue = toBN("2", 19);
```

Comment:
can you rename this to something understandable?  I guess that this is the `amountUnlockedPerBlock` right? what does pool means here? :confused:

Reply:
poolLockedValue = Value "Locked" for this test's pool and the same logic with poolBlockValue. But, okay, I renamed that